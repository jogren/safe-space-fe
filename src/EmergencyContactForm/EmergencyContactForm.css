.EmergencyContactForm {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  margin-top: 20px;
}

#emergency-contact-info-header {
  font-size: 30px;
  width: 75%;
}

#field {
  width: 50%;
  margin: auto;
  display: flex;
  justify-content: center;
  align-items: center;
}

#relationship {
  height: 50px;
  border-radius: 5px;
  margin-bottom: 10px;
  margin-left: 20px;
  width: 70%;
  font-size: 20px;
  padding-left: 5px;
}

#relationship-label {
  width: 45%;
  text-align: end;
  font-size: 20px;
}

#make-public-or-private-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  position: relative;
}

#public-or-private-button-text {
  width: 60%;
  font-size: 15px;
  text-align: center;
}

#make-public-or-private-button {
  width: 200px;
  height: 45px;
  font-size: 20px;
  background-color: #782a30;
  color: white;
  border-radius: 5px;
}

#make-public-or-private-container div {
  display: flex;
  justify-content: center;
  width: 250px;
  position: relative;
}

#make-public-or-private-container img {
  position: absolute;
  height: 20px;
  width: 20px;
  right: 0px;
  top: 12px;
}
