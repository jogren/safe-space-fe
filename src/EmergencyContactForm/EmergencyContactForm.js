import React from 'react'
import './EmergencyContactForm.css'

const EmergencyContactForm = () => {
  return (
    <form className="EmergencyContactForm">
      <h1>Emergency Contact</h1>

       <label for="Full Name">Name:</label>
       <input id="fullname" type="text" placeholder="Add Full Name"/>

       <label for="phone">Phone:</label>
       <input id="phone" type='text' placeholder="Add Phone Number"/>

       <label for="relationship">Relationship:</label>
       <input id="relationship" type='text' placeholder="Add relationship"/> 
    </form>
  )
}

export default EmergencyContactForm;