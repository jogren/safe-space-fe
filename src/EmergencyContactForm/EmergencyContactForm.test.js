import React from "react";
import { shallow } from "enzyme";
import EmergencyContactForm from "./EmergencyContactForm";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("EmergencyContactForm", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <EmergencyContactForm
        emergencyName=""
        setEmergencyName={jest.fn()}
        emergencyPhone=""
        setEmergencyPhone={jest.fn()}
        emergencyRelationship=""
        setEmergencyRelationship={jest.fn()}
        sendMessage={jest.fn()}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match the snapshot with three empty name, age, and phone inputs passed through", () => {
    expect(wrapper).toMatchSnapshot();
  });
});