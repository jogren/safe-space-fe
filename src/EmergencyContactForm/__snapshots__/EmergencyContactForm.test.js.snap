// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmergencyContactForm should match the snapshot with three empty name, age, and phone inputs passed through 1`] = `
<form
  className="EmergencyContactForm"
>
  <h1
    id="emergency-contact-info-header"
  >
    Emergency Contact Information:
  </h1>
  <div
    id="field"
  >
    <label
      htmlFor="Full Name"
      id="fullname-label"
    >
      Full Name:
    </label>
    <input
      id="fullname"
      name="name"
      onChange={[Function]}
      placeholder="Add Full Name"
      type="text"
      value=""
    />
  </div>
  <div
    id="field"
  >
    <label
      htmlFor="phone"
      id="phone-label"
    >
      Phone:
    </label>
    <input
      id="phone"
      name="phone"
      onChange={[Function]}
      placeholder="Add Phone Number"
      type="text"
      value=""
    />
  </div>
  <div
    id="field"
  >
    <label
      htmlFor="relationship"
      id="relationship-label"
    >
      Relationship:
    </label>
    <input
      id="relationship"
      name="relationship"
      onChange={[Function]}
      placeholder="Add Relationship"
      type="text"
      value=""
    />
  </div>
  <div
    id="make-public-or-private-container"
  >
    <h3
      id="public-or-private-button-text"
    >
      By clicking the following button, I agree to allow a message to be sent to the emergency contact number listed above.
    </h3>
    <button
      id="make-public-or-private-button"
      onClick={[Function]}
    >
      Send Text to Contact
    </button>
  </div>
</form>
`;
